### PR TITLE
Added .gitignore. Altered .gitlab-ci to cleanup after itself on every…

### PR DESCRIPTION
**Pull request by notjames**
… run. Also does a just-in-case cleanup in test for potential previous charts hanging around. A little duplicitous, but necessary for now

**What this PR does / why we need it**: Hopefully fixes an issue where if a previous build fails, the helm chart from the failed build doesn't linger, which causes all subsequent builds to fail CI.

### CODE REVIEW

**Comment by notjames** [comment]
self-merging as this is blocking. @alejandroEsc will verify later this evening.

**Comment by alejandroEsc** [comment]
Reverting due to questions.

**Comment by alejandroEsc** [comment]
Ill be ok to have this merged in again once i understand what the changes are/mean.

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,3 @@
-# ignore the generated Chart.yaml file
-vault-etcd/Chart.yaml
-
-# ignore the certs generation piece
-etcd-certs
```
Comment (alejandroEsc) [inline .gitignore]: please add this back. Certs are ignored and should not be pushed up the repo.

File: .gitignore [hunk 2]
```diff
@@ -42,3 +36,9 @@ docs/_build/
 # Mac
 .DS_Store
 
+# vim
+**.sw*
```
Comment (alejandroEsc) [inline .gitignore]: why these? Does not seem to be related to the code here?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -33,14 +40,12 @@ build:
 test:
   image: quay.io/samsung_cnct/kraken-tools:v0.8
   stage: test
-  variables:
```
Comment (alejandroEsc) [inline .gitlab-ci.yml]: why the move here? just want to make sure I understand this. these variables may not be used in all steps.

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -33,14 +40,12 @@ build:
...
-    NAMESPACE: ${APP_NAME}-test-${CI_PIPELINE_ID}
-    RELEASE: ${APP_NAME}-${CI_PIPELINE_ID}
-    DEPLOYMENT: ${APP_NAME}-${CI_PIPELINE_ID}-${APP_NAME}
   before_script:
   - mkdir /root/.kube
   - echo ${TEST_KUBECONFIG} | base64 -d > /root/.kube/config
   - helm init
+  - helm delete --purge ${RELEASE} || /bin/echo "${RELEASE} does not exist. This is good."
```
Comment (alejandroEsc) [inline .gitlab-ci.yml]: `This is good.` could you please remove this sentence. Also not sure what the intent here is?

File: .gitignore (hunk 2 above)
Comment (notjames) [reply .gitignore]: vim swap files.

File: .gitlab-ci.yml (hunk 3 above)
Comment (notjames) [reply .gitlab-ci.yml]: They are used in more than one step namely the cleanup steps.

File: .gitlab-ci.yml (hunk 4 above)
Comment (notjames) [reply .gitlab-ci.yml]: Don't want to remove the "this is good" line because to the lay-user, seeing this message /could cause alarm/. The intention here is to force clean the environment just before trying to use it. So for instance, in our use-case from yesterday, the cluster was "dirty" because your original helm delete failed to succeed. This item is specifically what fixed that, so this is kind of an edge case catcher.

File: .gitignore (hunk 1 above)
Comment (notjames) [reply .gitignore]: I'll add it back

File: .gitlab-ci.yml (hunk 4 above)
Comment (alejandroEsc) [reply .gitlab-ci.yml]: Ok can you please put instead `... environment is clean.` instead to make it a point that it was just checking that things were not dirty?

File: .gitlab-ci.yml (hunk 3 above)
Comment (alejandroEsc) [reply .gitlab-ci.yml]: Because of the clean up you mean?

File: .gitignore (hunk 2 above)
Comment (alejandroEsc) [reply .gitignore]: swap files are being pushed up? hmmmm weird.

Comment (notjames) [reply .gitignore]: No, they're not because I have to manually keep from adding them during my commits. This is a standard .gitignore entry for those of us who use vim to make it easy for us to do add/stage/commits

File: .gitlab-ci.yml (hunk 3 above)
Comment (notjames) [reply .gitlab-ci.yml]: yes

File: .gitlab-ci.yml (hunk 4 above)
Comment (notjames) [reply .gitlab-ci.yml]: Sure.

File: .gitlab-ci.yml (hunk 3 above)
Comment (alejandroEsc) [reply .gitlab-ci.yml]: sweet, makes sense, got it.
